#VRML_SIM R2021b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.007974429322973892 0.8859761455402486 0.463662245617019 3.1589320498487945
  position -0.10391076087071882 6.75326695009422 -4.181875421149685
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 5 5
  floorAppearance Parquetry {
  }
  wallHeight 0.0001
}
DEF Muro_Norte Solid {
  translation 0 0 2.48
  children [
    DEF Norte Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 5 0.3 0.1
      }
    }
  ]
  model "Muro Norte"
  boundingObject USE Norte
}
DEF Muro_Sur Solid {
  translation 0 0 -2.53
  children [
    DEF Sur Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 5 0.3 0.1
      }
    }
  ]
  name "solid(1)"
  model "Muro Norte"
  boundingObject USE Sur
}
DEF Muro_Este Solid {
  translation -2.45 0 -0.0599986
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF Este Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 5 0.3 0.1
      }
    }
  ]
  name "solid(2)"
  model "Muro Norte"
  boundingObject USE Este
}
DEF Muro_Oeste Solid {
  translation 2.46 0 -0.510004
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF Oeste Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 4 0.3 0.1
      }
    }
  ]
  name "solid(3)"
  model "Muro Norte"
  boundingObject USE Oeste
}
DEF Uno Solid {
  translation 1.52 0 2.28
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF Uno Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 1 0.3 0.1
      }
    }
  ]
  name "solid(16)"
  model "Muro Norte"
  boundingObject USE Uno
}
DEF Dos Solid {
  translation 0.52 0 0.93
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF Dos Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 1 0.3 0.1
      }
    }
  ]
  name "solid(4)"
  model "Muro Norte"
  boundingObject USE Dos
}
DEF Tres Solid {
  translation -1.48 0 0.95
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF Tres Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 1 0.3 0.1
      }
    }
  ]
  name "solid(8)"
  model "Muro Norte"
  boundingObject USE Tres
}
DEF Cuatro Solid {
  translation -1.48 0 -0.87
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF Cuatro Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 1 0.3 0.1
      }
    }
  ]
  name "solid(14)"
  model "Muro Norte"
  boundingObject USE Cuatro
}
DEF Cinco Solid {
  translation 1.45 0 -0.89
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF Cinco Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 1 0.3 0.1
      }
    }
  ]
  name "solid(12)"
  model "Muro Norte"
  boundingObject USE Cinco
}
DEF Seis Solid {
  translation -0.48 0 0.03
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF Seis Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 1 0.3 0.1
      }
    }
  ]
  name "solid(10)"
  model "Muro Norte"
  boundingObject USE Seis
}
DEF Siete Solid {
  translation 1.5 0 0.48
  rotation 0 1 0 3.14159
  children [
    DEF Siete Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 2 0.3 0.1
      }
    }
  ]
  name "solid(6)"
  model "Muro Norte"
  boundingObject USE Siete
}
DEF Ocho Solid {
  translation 0.0686 0 1.43
  rotation 0 1 0 0.02
  children [
    DEF Ocho Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 1 0.3 0.1
      }
    }
  ]
  name "solid(7)"
  model "Muro Norte"
  boundingObject USE Ocho
}
DEF Nueve Solid {
  translation -1.98 0 -0.42
  rotation 0 1 0 3.14159
  children [
    DEF Nueve Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 1 0.3 0.1
      }
    }
  ]
  name "solid(15)"
  model "Muro Norte"
  boundingObject USE Nueve
}
DEF Diez Solid {
  translation -1.5 0 0.48
  rotation 0 1 0 3.14159
  children [
    DEF Diez Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 2 0.3 0.1
      }
    }
  ]
  name "solid(9)"
  model "Muro Norte"
  boundingObject USE Diez
}
DEF Once Solid {
  translation 0.5 0 -0.42
  rotation 0 1 0 3.14159
  children [
    DEF Once Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 2 0.3 0.1
      }
    }
  ]
  name "solid(11)"
  model "Muro Norte"
  boundingObject USE Once
}
DEF Doce Solid {
  translation -0.53 0 -1.4
  rotation 0 1 0 3.14159
  children [
    DEF Doce Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 2 0.3 0.1
      }
    }
  ]
  name "solid(13)"
  model "Muro Norte"
  boundingObject USE Doce
}
DEF Start Solid {
  translation 2.46 0 1.96
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF Start Shape {
      appearance PBRAppearance {
        baseColor 0.666667 0 0
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 1.01 0.301 0.101
      }
    }
  ]
  name "solid(5)"
  model "Muro Norte"
  boundingObject USE Start
}
DEF End Solid {
  translation -2.45 0 0.03
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF Start Shape {
      appearance PBRAppearance {
        baseColor 0.666667 0 0
        baseColorMap ImageTexture {
        }
      }
      geometry Box {
        size 1.01 0.301 0.101
      }
    }
  ]
  name "solid(17)"
  model "Muro Norte"
  boundingObject USE Start
}
DEF Robot_Brazo Robot {
  translation 2.05 0.038 2.15
  rotation 0 1 0 3.141592653589793
  children [
    DEF Sensor_Adelante DistanceSensor {
      translation 0 0 -0.08
      rotation 0 1 0 1.5708003061004252
      name "Sensor_Delantero"
      lookupTable [
        0 0 0
        0.2 1500 0
      ]
    }
    DEF Sensor_Atras DistanceSensor {
      translation 0 0 0.08
      rotation 0 1 0 -1.5708003061004252
      name "Sensor_Trasero"
      lookupTable [
        0 0 0
        0.2 1500 0
      ]
    }
    DEF Sensor_Izquierda DistanceSensor {
      translation -0.07 0 0
      rotation 0 1 0 -3.141592653589793
      name "Sensor_Izquierdo"
      lookupTable [
        0 0 0
        0.2 1500 0
      ]
      type "sonar"
    }
    DEF Sensor_Derecha DistanceSensor {
      translation 0.07 0 0
      name "Sensor_Derecho"
      lookupTable [
        0 0 0
        0.2 1500 0
      ]
    }
    DEF Base_Robot HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor 0 0.028 0
      }
      device [
        PositionSensor {
          name "Encoder_Base"
        }
        RotationalMotor {
          name "Motor_Base"
        }
      ]
      endPoint Solid {
        translation 0 0.028 0
        rotation 0 -1 0 0
        children [
          DEF Muesca Transform {
            translation -0.022 0 0
            children [
              DEF Muesca Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                }
                geometry Box {
                  size 0.06 0.0051 0.005
                }
              }
            ]
          }
          DEF Art_eslabones_1 Transform {
            translation 0 0.01 0
            children [
              DEF Art_eslabones_1 Shape {
                appearance PBRAppearance {
                  baseColor 0.666667 0 1
                }
                geometry Sphere {
                  radius 0.016
                }
              }
            ]
          }
          DEF Eslabon_1 Hinge2Joint {
            jointParameters HingeJointParameters {
              axis -1 0 0
              anchor 0 0.01 0
            }
            jointParameters2 JointParameters {
              axis 0 0 -1
            }
            device [
              DEF Motor_Eslabon_1 RotationalMotor {
                name "Motor_Eslabon_1_x"
              }
              DEF Encoder_Eslabon_1 PositionSensor {
                name "Encoder_Eslabon_1_x"
              }
            ]
            device2 [
              DEF Motor_Eslabon_1_z RotationalMotor {
                name "Motor_Eslabon_1_z"
              }
              DEF Encoder_Eslabon_1_z PositionSensor {
                name "Encoder_Eslabon_1_z"
              }
            ]
            endPoint Solid {
              translation -9.430478704025571e-12 0.06999999999999997 -4.1359030627651384e-25
              children [
                DEF Art_escalones_2 HingeJoint {
                  jointParameters HingeJointParameters {
                    anchor 0 0.06 0
                  }
                  device [
                    RotationalMotor {
                      name "Motor_Esfera_2"
                    }
                  ]
                  endPoint Solid {
                    translation 0 0.06 0
                    rotation 1 0 0 0
                    children [
                      DEF Eslabon_2 Hinge2Joint {
                        jointParameters HingeJointParameters {
                        }
                        jointParameters2 JointParameters {
                        }
                        device [
                          RotationalMotor {
                            name "Motor_Eslabon_2_x"
                          }
                          DEF Encoder_Eslabon_1_x PositionSensor {
                            name "Encoder_Eslabon_2_x"
                          }
                        ]
                        device2 [
                          DEF Encoder_Eslabon_2_z PositionSensor {
                            name "Encoder_Eslabon_2_z"
                          }
                          RotationalMotor {
                            name "Motor_Eslabon_2_z"
                          }
                        ]
                        endPoint Solid {
                          translation 5.2908816564684115e-11 0.060000000000000026 0
                          children [
                            DEF Muesca Transform {
                              translation -0.016 0 0
                              children [
                                DEF Muesca Shape {
                                  appearance PBRAppearance {
                                    baseColor 0 0 1
                                  }
                                  geometry Box {
                                    size 0.001 0.1 0.005
                                  }
                                }
                              ]
                            }
                            DEF Eslabon_2 Shape {
                              appearance PBRAppearance {
                                baseColor 0.666667 0 1
                              }
                              geometry Cylinder {
                                height 0.1
                                radius 0.016
                              }
                            }
                          ]
                          boundingObject DEF Eslabon_2 Shape {
                            appearance PBRAppearance {
                              baseColor 0.666667 0 1
                            }
                            geometry Cylinder {
                              height 0.1
                              radius 0.016
                            }
                          }
                          physics Physics {
                          }
                        }
                      }
                      DEF Art_eslabones_2 Shape {
                        appearance PBRAppearance {
                          baseColor 0.666667 0 1
                        }
                        geometry Sphere {
                          radius 0.016
                        }
                      }
                    ]
                    boundingObject USE Art_eslabones_2
                    physics Physics {
                    }
                    recognitionColors [
                      0 0 0
                    ]
                  }
                }
                DEF Eslabon_1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.666667 0 1
                  }
                  geometry Cylinder {
                    height 0.1
                    radius 0.016
                  }
                }
                DEF Muesca Transform {
                  translation -0.016 0 0
                  children [
                    DEF Muesca Shape {
                      appearance PBRAppearance {
                        baseColor 0 0 1
                      }
                      geometry Box {
                        size 0.001 0.1 0.005
                      }
                    }
                  ]
                }
              ]
              name "solid(7)"
              boundingObject USE Eslabon_1
              physics Physics {
              }
            }
          }
          DEF Base_Robot Shape {
            appearance PBRAppearance {
              baseColor 0.666667 0 0.498039
            }
            geometry Cylinder {
              height 0.005
              radius 0.06
              subdivision 6
            }
          }
        ]
        boundingObject USE Base_Robot
        physics Physics {
        }
      }
    }
    DEF Rueda_1 HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 -1
        anchor -0.075 -0.013 -0.075
      }
      device [
        DEF Encoder_1 PositionSensor {
          name "Encoder_1"
        }
        DEF Motor_1 RotationalMotor {
          name "Motor_1"
        }
      ]
      endPoint Solid {
        translation -0.075 -0.013 -0.075
        rotation 0.7071067811865475 -1.9624714013651983e-17 -0.7071067811865476 1.570943261201891
        children [
          DEF Muesca Transform {
            rotation 0 1 0 -0.7853996938995746
            children [
              DEF Muesca Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                }
                geometry Box {
                  size 0.048 0.0105 0.01
                }
              }
              DEF Parte_Superior Transform {
                translation -0.019 0 0
                children [
                  DEF Parte_Superior Shape {
                    appearance PBRAppearance {
                      baseColor 0 0.333333 0
                    }
                    geometry Box {
                      size 0.01051 0.01051 0.011
                    }
                  }
                ]
              }
            ]
          }
          DEF RDI Shape {
            appearance PBRAppearance {
              baseColor 0.666667 0 1
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "solid(3)"
        boundingObject USE RDI
        physics Physics {
        }
      }
    }
    DEF Rueda_2 HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 1
        anchor 0.075 -0.013 -0.075
      }
      device [
        DEF Encoder_2 PositionSensor {
          name "Encoder_2"
        }
        DEF Motor_2 RotationalMotor {
          name "Motor_2"
        }
      ]
      endPoint Solid {
        translation 0.075 -0.013 -0.075
        rotation 0.7071067811865476 0 0.7071067811865476 1.5707996938995743
        children [
          DEF Muesca Transform {
            rotation 0 1 0 -2.3562003061004253
            children [
              DEF Muesca Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                }
                geometry Box {
                  size 0.048 0.0105 0.01
                }
              }
              DEF Parte_Superior Transform {
                translation -0.019 0 0
                children [
                  DEF Parte_Superior Shape {
                    appearance PBRAppearance {
                      baseColor 0 0.333333 0
                    }
                    geometry Box {
                      size 0.01051 0.01051 0.011
                    }
                  }
                ]
              }
            ]
          }
          DEF RDI Shape {
            appearance PBRAppearance {
              baseColor 0.666667 0 1
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "solid(2)"
        boundingObject USE RDI
        physics Physics {
        }
      }
    }
    DEF Rueda_3 HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 1
        anchor -0.075 -0.013 0.075
      }
      device [
        DEF Encoder_3 PositionSensor {
          name "Encoder_3"
        }
        DEF Motor_3 RotationalMotor {
          name "Motor_3"
        }
      ]
      endPoint Solid {
        translation -0.075 -0.013 0.075
        rotation 0.7071066717798298 1.9626122691970244e-17 0.7071068905932484 1.5707996938995747
        children [
          DEF Muesca Transform {
            rotation 0 1 0 -2.3561996938995744
            children [
              DEF Muesca Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                }
                geometry Box {
                  size 0.048 0.0105 0.01
                }
              }
              DEF Parte_Superior Transform {
                translation -0.019 0 0
                children [
                  DEF Parte_Superior Shape {
                    appearance PBRAppearance {
                      baseColor 0 0.333333 0
                    }
                    geometry Box {
                      size 0.01051 0.01051 0.011
                    }
                  }
                ]
              }
            ]
          }
          DEF RDI Shape {
            appearance PBRAppearance {
              baseColor 0.666667 0 1
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "solid(5)"
        boundingObject USE RDI
        physics Physics {
        }
      }
    }
    DEF Rueda_4 HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 -1
        anchor 0.075 -0.013 0.075
      }
      device [
        DEF Encoder_4 PositionSensor {
          name "Encoder_4"
        }
        DEF Motor_4 RotationalMotor {
          name "Motor_4"
        }
      ]
      endPoint Solid {
        translation 0.075 -0.013 0.075
        rotation 0.7071067811865477 -1.9626119688198786e-17 -0.7071067811865476 1.5707999999999998
        children [
          DEF Muesca Transform {
            rotation 0 1 0 -0.7853996938995746
            children [
              DEF Muesca Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                }
                geometry Box {
                  size 0.048 0.0105 0.01
                }
              }
              DEF Parte_Superior Transform {
                translation -0.019 0 0
                children [
                  DEF Parte_Superior Shape {
                    appearance PBRAppearance {
                      baseColor 0 0.333333 0
                    }
                    geometry Box {
                      size 0.01051 0.01051 0.011
                    }
                  }
                ]
              }
            ]
          }
          DEF RDI Shape {
            appearance PBRAppearance {
              baseColor 0.666667 0 1
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "solid(1)"
        boundingObject USE RDI
        physics Physics {
        }
      }
    }
    DEF Cuerpo Transform {
      children [
        DEF Cuerpo_mesh Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.14 0.05 0.14
          }
        }
        DEF Muesca_Cuerpo Transform {
          translation -0.06 0 -0.06
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 0 0
              }
              geometry Box {
                size 0.01 0.051 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  boundingObject USE Cuerpo_mesh
  physics Physics {
  }
  controller "Roberto_Robato_Laberinto"
}
